lib/checkunusedfunctions.cpp

```diff
@@ -231,6 +231,55 @@ void CheckUnusedFunctions::parseTokens(const Tokenizer &tokenizer, const char Fi
 }
 
 
+static bool isOperatorFunction(const std::string & funcName) {
+    /* Operator functions are invalid function names for C, so no need to check
+     * this in here. As result the returned error function might be incorrect.
+     *
+     * List of valid operators can be found at:
+     * http://en.cppreference.com/w/cpp/language/operators
+     *
+     * Conversion functions must be a member function (at least for gcc), so no
+     * need to cover them for unused functions.
+     *
+     * To speed up the comparision, not the whole list of operators is used.
+     * Instead only the character after the operator prefix is checked to be a
+     * none alpa numeric value, but the '_', to cover function names like
+     * "operator_unused". In addition the following valid operators are checked:
+     * - new
+     * - new[]
+     * - delete
+     * - delete[]
+     */
+    const std::string operatorPrefix = "operator";
+    if (funcName.compare(0, operatorPrefix.length(), operatorPrefix) != 0) {
+        return false;
+    }
+
+    const char firstOperatorChar = funcName[operatorPrefix.length()];
+    if (firstOperatorChar == '_') {
+        return false;
+    }
+
+    if (!std::isalnum(firstOperatorChar)) {
+        return true;
+    }
+
+    const std::vector<std::string> additionalOperators = make_container< std::vector<std::string> >()
+        << "new"
+        << "new[]"
+        << "delete"
+        << "delete[]";
+    const std::string opName = funcName.substr(operatorPrefix.length());
+
+    for (std::vector<std::string>::const_iterator curOp = additionalOperators.begin(); curOp != additionalOperators.end(); ++curOp) {
```

Comment:
Using  `std::find()`, you only need 1 line of code to return true if vector has item.